src/Symfony/Component/DependencyInjection/Tests/Compiler/RemoveUnusedDefinitionsPassTest.php

```diff
@@ -86,14 +86,12 @@ public function testProcessWontRemovePrivateFactory()
 
         $container
             ->register('foo', 'stdClass')
-            ->setFactoryClass('stdClass')
-            ->setFactoryMethod('getInstance')
+            ->setFactory(array('stdClass', 'getInstance'))
             ->setPublic(false);
 
         $container
             ->register('bar', 'stdClass')
-            ->setFactoryService('foo')
-            ->setFactoryMethod('getInstance')
+            ->setFactory(array(new Reference('foo'), 'getInstance'))
```

Comment:
We need to keep legacy tests ensuring that it still works for definitions using the deprecated way